### PR TITLE
Add distributed JMeter chart

### PR DESCRIPTION
**Pull request by pedrocesarti**
- Creates deployment resource for the JMeter master and server
- Creates service resource for JMeter server

Signed-off-by: Pedro César de Azevedo <pedrocesar.ti@gmail.com>

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://www.helm.sh/blog/helm-dco/index.html) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md

### CODE REVIEW

**Comment by cpanato** [comment]
@pedrocesar-ti 
```
stable/distributed-jmeter/Chart.yaml
  13:1      error    too many blank lines (1 > 0)  (empty-lines)
```

**Comment by pedrocesar-ti** [comment]
Fixed @cpanato

File: stable/distributed-jmeter/templates/jmeter-master-deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,32 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ .Release.Name }}-master
```
Comment (unguiculus) [inline stable/distributed-jmeter/templates/jmeter-master-deployment.yaml]: Resoure names should use the fullname template.

File: stable/distributed-jmeter/templates/jmeter-master-deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+      labels:
+        app.kubernetes.io/name:  {{ include "distributed-jmeter.name" . }}
+        app.kubernetes.io/instance: {{ .Release.Name }}-master
+        role: master
+    spec:
+      containers:
+        - name: {{ .Chart.Name }}
+          image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
+          args: ["master"]
```
Comment (unguiculus) [inline stable/distributed-jmeter/templates/jmeter-master-deployment.yaml]: Please don't mix list syntaxes. Use this:

```yaml
args: 
  - master
```

**Review by unguiculus — CHANGES_REQUESTED** [review]

**Review by unguiculus — APPROVED** [review]

File: stable/distributed-jmeter/templates/jmeter-master-deployment.yaml (hunk 1 above)
Comment (cpanato) [reply stable/distributed-jmeter/templates/jmeter-master-deployment.yaml]: @pedrocesar-ti can you please apply the changes?

File: stable/distributed-jmeter/Chart.yaml [hunk 3]
```diff
@@ -0,0 +1,12 @@
+apiVersion: v1
+appVersion: "1.0"
```
Comment (cpanato) [inline stable/distributed-jmeter/Chart.yaml]: appVersion should match the tag that you are using in this case: `tag: 3.3`

**Review by cpanato — CHANGES_REQUESTED** [review]

**Review by cpanato — APPROVED** [review]
lgtm

**Review by unguiculus — CHANGES_REQUESTED** [review]
I'd suggest you go for `app.kubernetes.io/component` instead of `role`. See https://github.com/helm/helm/blob/master/docs/chart_best_practices/labels.md

File: stable/distributed-jmeter/.helmignore [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+.svn/
+# Common backup files
+*.swp
+*.bak
+*.tmp
+*~
+# Various IDEs
+.project
+.idea/
+*.tmproj
```
Comment (cpanato) [inline stable/distributed-jmeter/.helmignore:21]: Please add the OWNERS file here

File: stable/distributed-jmeter/templates/jmeter-master-deployment.yaml [hunk 5]
```diff
@@ -0,0 +1,33 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ template "distributed-jmeter.fullname" . }}-master
+  labels:
+    app.kubernetes.io/name: {{ include "distributed-jmeter.name" . }}
+    helm.sh/chart: {{ include "distributed-jmeter.chart" . }}
+    app.kubernetes.io/instance: {{ .Release.Name }}
+    role: master
```
Comment (cpanato) [inline stable/distributed-jmeter/templates/jmeter-master-deployment.yaml]: missing `app.kubernetes.io/managed-by`

File: stable/distributed-jmeter/templates/jmeter-server-deployment.yaml [hunk 6]
```diff
@@ -0,0 +1,33 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ template "distributed-jmeter.fullname" . }}-server
+  labels:
+    app.kubernetes.io/name: {{ include "distributed-jmeter.name" . }}
+    helm.sh/chart: {{ include "distributed-jmeter.chart" . }}
+    app.kubernetes.io/instance: {{ .Release.Name }}
```
Comment (cpanato) [inline stable/distributed-jmeter/templates/jmeter-server-deployment.yaml:8]: missing `app.kubernetes.io/managed-by`

File: stable/distributed-jmeter/templates/jmeter-master-deployment.yaml [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+spec:
+  replicas: {{ .Values.master.replicaCount }}
+  strategy:
+    type: RollingUpdate
+  selector:
+    matchLabels:
+      app.kubernetes.io/name: {{ include "distributed-jmeter.name" . }}
+      app.kubernetes.io/instance: {{ .Release.Name }}-master
```
Comment (cpanato) [inline stable/distributed-jmeter/templates/jmeter-master-deployment.yaml]: you can remove the `-master` dont need

File: stable/distributed-jmeter/templates/jmeter-master-deployment.yaml [hunk 8]
```diff
@@ -0,0 +1,34 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ template "distributed-jmeter.fullname" . }}-master
+  labels:
+    app.kubernetes.io/name: {{ include "distributed-jmeter.name" . }}
+    helm.sh/chart: {{ include "distributed-jmeter.chart" . }}
+    app.kubernetes.io/instance: {{ .Release.Name }}
+    app.kubernetes.io/managed-by: {{ .Release.Service }}
+    role: master
```
Comment (unguiculus) [inline stable/distributed-jmeter/templates/jmeter-master-deployment.yaml]: Please use `app.kubernetes.io/component` instead of `role`.

**Review by unguiculus — CHANGES_REQUESTED** [review]

File: stable/distributed-jmeter/templates/jmeter-master-deployment.yaml (hunk 8 above)
Comment (pedrocesarti) [reply stable/distributed-jmeter/templates/jmeter-master-deployment.yaml]: Done!

**Pull request merged**